File: js/src/sorting/Selection.js
import Sorter from './Sorter';

class Selection extends Sorter {
    constructor(arr = []) {
        super(arr);
    }

    sort() {
        let N = this.arr.length;
        for (let i = 0; i < N; i++) {
            let min = i;
            for (let j = i+1; j < N; j++) {
                if (this.less(this.arr[j], this.arr[min])) {
                    min = j;
                }
            }

            this.exch(i, min);
        }
    }
}

module.exports = Selection;

File: java/src/com/test/sorting/Selection.java
package com.test.sorting;


public class Selection extends Sorter
{
    public Selection(int[] arr)
    {
        super(arr);
    }

    @Override
    public void sort()
    {
        int N = arr.length;
        for (int i = 0; i < N; i++) {
            int min = i;
            for (int j = i+1; j < N; j++) {
                if (less(arr[j], arr[min])) {
                    min = j;
                }
            }

            exch(i, min);
        }
    }
}
